lib/lambda_ethereum_consensus/state_transition/misc.ex

```diff
@@ -12,6 +13,69 @@ defmodule LambdaEthereumConsensus.StateTransition.Misc do
     div(slot, slots_per_epoch)
   end
 
+  @doc """
+     Return the shuffled index corresponding to ``seed`` (and ``index_count``).
+  """
+  @spec compute_shuffled_index(SszTypes.uint64(), SszTypes.uint64(), SszTypes.bytes32()) ::
+          {:ok, SszTypes.uint64()} | {:error, String.t()}
+  def compute_shuffled_index(index, index_count, seed) do
+    result =
+      unless index < index_count and index_count > 0 do
+        {:error, "index not less than index count"}
+      else
+        shuffle_round_count = ChainSpec.get("SHUFFLE_ROUND_COUNT")
+
+        new_index =
+          Enum.reduce(0..(shuffle_round_count - 1), index, fn round, current_index ->
+            round_as_bytes =
+              <<round>> |> :binary.decode_unsigned() |> :binary.encode_unsigned(:little)
+
+            seed_as_bytes = seed |> :binary.decode_unsigned() |> :binary.encode_unsigned(:little)
+
+            hash_of_seed_round =
+              :crypto.hash(:sha256, seed_as_bytes <> round_as_bytes)
+              |> :binary.decode_unsigned()
+              |> :binary.encode_unsigned(:little)
+
+            first_8_bytes_of_hash_of_seed_round =
+              hash_of_seed_round |> :binary.bin_to_list({0, 8}) |> :binary.list_to_bin()
```

Comment:
I suggest reading about binary pattern matching with `<<>>`: https://hexdocs.pm/elixir/1.12/Kernel.SpecialForms.html#%3C%3C%3E%3E/1
```suggestion
            <<first_8_bytes_of_hash_of_seed_round::binary-size(8), _::binary>> = hash_of_seed_round
```